exp/runtime/internal/controllers/extension_controller.go

```diff
@@ -69,14 +67,6 @@ func (r *ExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
```

Comment:
Not sure how far we want to go in this PR.

But can you maybe add that line already in ~ l.61 so we don't forget it later
```go
r.Registry.RemoveRuntimeExtension(ext)
```

I did some tests in another context and turns out that's the case we hit when a resource is deleted.
(prior art: ClusterCacheReconciler)

Reply:
I guess we can drop the reconcileDelete for now until we're taking a closer look at registry cleanup for delete. But the remove in l.63 should cover 99% of the cases

Reply:
The only changes made in the controller here were to prevent build errors, (and quickly at that!). I'll do a follow up PR in the reconciler.

Reply:
Got it